### PR TITLE
Base64 Links Decoder Plugin

### PR DESCRIPTION
**Pull request by nawelasg**
This Plugins decodes the base64 links in the client itself, It has a dedicated UI button on the ChatButtons and Open a separate Modal window to display the decoded base64 string.

### CODE REVIEW

File: src/plugins/decodeBase64Links/index.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                </ModalFooter>
+            </ModalRoot>
+        </ErrorBoundary>
+    ));
+};
+
+export default definePlugin({
+    name: "DecodeBase64Links",
+    description: "Decode base64 content of any selected text in a message.",
+    authors: [{ name: "Nawelasg", id: 0n }],
```
Comment (nin0-dev) [inline src/plugins/decodeBase64Links/index.tsx:85]: for your next plugins, add yourself to the Devs object instead of putting the author inline

**Review by nin0-dev — COMMENTED** [review]
this can easily be done with a user app, i doubt this will be merged

File: src/plugins/decodeBase64Links/index.tsx (hunk 1 above)
Comment (nawelasg) [reply src/plugins/decodeBase64Links/index.tsx]: I have resolved the Devs object issue in the new commit.

**Comment by nawelasg** [comment]
> this can easily be done with a user app, i doubt this will be merged

But it will more useful as a plugin in vencord, because this operates only on the client side and has nothing to do with discord. 
Whereas if its an user app or a bot, Discord will be able to see what is being shared and decoded. And the reason why someone encode such stuff, There is a reason they dont want discord to be able to see that stuff.

**Comment by Koishukaze** [comment]
it's not private either way, anyone can easily decode it

**Comment by nawelasg** [comment]
But its not directly under discord surveillance. Thats what it achieves.

**Comment by nin0-dev** [comment]
i don't think that you have understood that for you to even get the b64 message to decrypt

you...

need to make it go through discord servers

this plugin isn't achieving anything

**Comment by Vendicated** [comment]
this is
a) not useful
b) a duplicate
c) better done via a bot
d) base64 achieves nothing for privacy. use an end to end encrypted messenger like signal, matrix, xmpp, etc

**Pull request closed without merging**